xcell/mappers/mapper_WIxSC.py

```diff
@@ -0,0 +1,248 @@
+           'mask': 'mask.fits',
+           'star_map': 'stars.fits',
+           'spec_sample': 'zSpec-comp-WIxSC.csv',
+           'bin_name': '0',
+           'z_edges': [0, 0.5],
+           'path_rerun': '.',
+           'n_jk_dir': 100,
+           'mask_name': 'mask_WIxSC'}
+        """
+        self._get_defaults(config)
+        self.z_edges = config.get('z_edges', [0, 0.5])
+
+        self.cat_data = None
+        self.npix = hp.nside2npix(self.nside)
+        self.bn = self.config['bin_name']
+
+        # Angular mask
+        self.dndz = None
+        self.mask = None
+        self.stars = None
+        self.dndz = None
+        self.delta_map = None
+        self.nl_coupled = None
+        self.mask = None
+        self.nside_nl_threshold = config.get('nside_nl_threshold',
+                                             4096)
+        self.lmin_nl_from_data = config.get('lmin_nl_from_data',
+                                            2000)
+
+    def get_catalog(self):
+        if self.cat_data is None:
+            fn = 'WIxSC_lite_bin' + self.bn + '.fits'
+            f_exists, fname_lite = self._check_rerun_file_exists(fn)
+            # Check if lite catalog exists
+            if f_exists:
+                with fits.open(fname_lite) as f:
+                    self.cat_data = Table.read(f, format='fits', memmap=True)
+            else:
+                file_data = self.config['data_catalog']
+                if not os.path.isfile(file_data):
+                    raise ValueError(f"File {file_data} not found")
+                # Read catalog
+                with fits.open(file_data) as f:
+                    self.cat_data = Table.read(f, format='fits', memmap=True)
+                # Bin in redshift
+                self.cat_data = self._bin_z(self.cat_data)
+                # Ditch useless columns
+                self.cat_data.keep_columns(['RA', 'DEC',
+                                            'W1MCORR', 'W2MCORR',
+                                            'RCALCORR', 'BCALCORR'])
+                # Sky mask
+                self.cat_data = self._mask_catalog(self.cat_data)
+                # Save lite if needed
+                if fname_lite is not None:
+                    self.cat_data.write(fname_lite)
+        return self.cat_data
+
+    def _check_rerun_file_exists(self, fname):
+        path_i = self.config.get('path_rerun', None)
+        if path_i is None:
+            return False, None
+        else:
+            fname_full = os.path.join(path_i, fname)
+            return os.path.isfile(fname_full), fname_full
+
+    def _mask_catalog(self, cat):
+        self.mask = self.get_mask()
+        ipix = hp.ang2pix(self.nside,
+                          np.degrees(cat['RA']),
+                          np.degrees(cat['DEC']),
+                          lonlat=True)
+        # Mask is binary, so 0.1 or 0.00001 doesn't really matter
+        return cat[self.mask[ipix] > 0.1]
+
+    def _bin_z(self, cat):
```

Comment:
I don't know if it matters but in KiDS mapper we used the convention z_min < z <= z_max

Reply:
good, let's be consistent. Fixed